### ISSUE TITLE
"required" property should be a must-have

### ISSUE THREAD

**Issue by manudiv** [mozilla/node-convict#391]

**Comment by AaronTorgerson** [comment]
Agreed. It is rather odd to not be able to specify that "this property must be set from either a file or environment variable - no default is allowable".

**Comment by ath88** [comment]
I would love this feature.

**Comment by arminrosu** [comment]
Additionally, the current suggested way of setting `default: null` and `format: String` properties means that typings infer the values as `null | undefined`. Thus your build will fail if using Typescript, though on runtime only `.validate()` will throw an error.
